src/error.ts

```diff
@@ -1,111 +1,123 @@
-  ERR_CONNECTION_CLOSED = 'ERR_CONNECTION_CLOSED',
 }
 
 export class ConnectionClosedError extends WebRTCTransportError {
-  constructor(state: RTCPeerConnectionState, msg: string) {
-    super(`peerconnection moved to state: ${state}:` + msg);
-    this.name = 'WebRTC/ConnectionClosed';
+  constructor (state: RTCPeerConnectionState, msg: string) {
+    super(`peerconnection moved to state: ${state}:` + msg)
+    this.name = 'WebRTC/ConnectionClosed'
   }
 }
 
-export function connectionClosedError(state: RTCPeerConnectionState, msg: string) {
-  return createError(new ConnectionClosedError(state, msg), codes.ERR_CONNECTION_CLOSED)
+export function connectionClosedError (state: RTCPeerConnectionState, msg: string) {
+  return errCode(new ConnectionClosedError(state, msg), codes.ERR_CONNECTION_CLOSED)
 }
 
 export class InvalidArgumentError extends WebRTCTransportError {
-  constructor(msg: string) {
-    super('There was a problem with a provided argument: ' + msg);
-    this.name = 'WebRTC/InvalidArgumentError';
+  constructor (msg: string) {
+    super('There was a problem with a provided argument: ' + msg)
+    this.name = 'WebRTC/InvalidArgumentError'
   }
 }
 
-export function unsupportedHashAlgorithm(algorithm: string) {
-  return createError(new UnsupportedHashAlgorithmError(algorithm), codes.ERR_HASH_NOT_SUPPORTED);
+export function unsupportedHashAlgorithm (algorithm: string) {
+  return errCode(new UnsupportedHashAlgorithmError(algorithm), codes.ERR_HASH_NOT_SUPPORTED)
 }
 
 export class UnsupportedHashAlgorithmError extends WebRTCTransportError {
-  constructor(algo: string) {
-    let msg = `unsupported hash algorithm: ${algo}`;
-    super(msg);
-    this.name = 'WebRTC/UnsupportedHashAlgorithmError';
+  constructor (algo: string) {
+    const msg = `unsupported hash algorithm: ${algo}`
+    super(msg)
+    this.name = 'WebRTC/UnsupportedHashAlgorithmError'
   }
 }
 
-export function invalidArgument(msg: string) {
-  return createError(new InvalidArgumentError(msg), codes.ERR_INVALID_PARAMETERS);
+export function invalidArgument (msg: string) {
+  return errCode(new InvalidArgumentError(msg), codes.ERR_INVALID_PARAMETERS)
 }
 
 export class UnimplementedError extends WebRTCTransportError {
-  constructor(methodName: string) {
-    super('A method (' + methodName + ') was called though it has been intentionally left unimplemented.');
-    this.name = 'WebRTC/UnimplementedError';
+  constructor (methodName: string) {
+    super('A method (' + methodName + ') was called though it has been intentionally left unimplemented.')
+    this.name = 'WebRTC/UnimplementedError'
+  }
+}
+
+export function unimplemented (methodName: string) {
+  return errCode(new UnimplementedError(methodName), codes.ERR_NOT_IMPLEMENTED)
+}
+
+export class InvalidFingerprintError extends WebRTCTransportError {
+  constructor (fingerprint: string, source: string) {
+    super(`Invalid fingerprint "${fingerprint}" within ${source}`)
```

Comment:
For security reasons one does usually not add such details to an error, even though it might be UX. Dunno. Up to you.

Reply:
@TrophyNinjaShrub @ckousik thoughts?

Reply:
The fingerprint is public in the multiaddr and cannot reveal any information not available to an attacker.